### PR TITLE
Bump go 1.22.2

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

- Bump go to v 1.22.2 and the appropriate changes in cloudbuilder and go linter
- Refactor simple loops to loop over integers where possible
- Fix a test to use the cluster exposed in the CLUSTER_NAME env var rather than the hardcoded kind cluster

- Fix glooctl failing tests : Bump the istio version used in the kube2e tests to 1.19.9 that support k8s 1.28 and a higher version of envoy (https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases)
Using a lower version that did not support k8s 1.28 or envoy 1.29  led to test failures as seen [here](https://github.com/solo-io/gloo/actions/runs/8664592439/job/23761367282?pr=9350)
Ideally we should use v1.21 that supports envoy v1.29 (istio 1.19 supports envoy 1.27) but that might require additional changes that can be handled in a separate PR

Enterprise PR with the gloo bump - https://github.com/solo-io/solo-projects/pull/6025

## CI changes
- Bump cloudbuilder and go linter

## Docs changes
N/A

# Context
https://github.com/solo-io/gloo/issues/8981

## Interesting decisions
Chose to refactor simple loops to loop over integers as introduced in go 1.21 instead of the old way to prettify the code

## Testing steps
Tests passing

## Notes for reviewers
Ensure that only loops iterating from 0 to n with single increments have been changed. ie: `i := 0; i < num; i++`

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
     if: ${{ !github.event.pull_request.draft }}
     steps:
     - uses: actions/checkout@v4
-    - name: Setup Go 1.21
+    - name: Setup Go
       uses: actions/setup-go@v4
       with:
-        go-version: "1.21"
+        go-version-file: go.mod
```
Comment (nfuden) [inline .github/workflows/pr.yaml:17]: any objections to this @jenshu

Comment (jenshu) [reply .github/workflows/pr.yaml:17]: this should be fine

**Review by nfuden — DISMISSED** [review]
nice stuff.
I did not explicitly do go build -gcflags=all=-d=loopvar=2 cmd/go but overall looks good

**Comment by davidjumani** [comment]
> I did not explicitly do go build -gcflags=all=-d=loopvar=2 cmd/go but overall looks good

It was run as until https://github.com/solo-io/gloo/pull/9342/commits/2df26065c5a6cc9f84c712a6393c6a23e0292b21 without any warnings and removed before marking it as ready

File: cloudbuild-cache.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 options:
   env:
-    - "_GO_VERSION=1.21.5"
+    - "_GO_VERSION=1.22.1"
```
Comment (ben-taussig-solo) [inline cloudbuild-cache.yaml:3]:
```suggestion
    - "_GO_VERSION=1.22.2"
```

**Review by ben-taussig-solo — APPROVED** [review]

**Review by nfuden — APPROVED** [review]
we may want to look into istio failing rationale at some point but getting 1.22 in is pretty good so 🚀

**Pull request merged**